src/Groups/group_constructors.jl

```diff
@@ -843,38 +845,81 @@ julia> relators(g)
  r^2*s^-2
  s^4
  r^-1*s*r*s
-
 ```
 """
 quaternion_group(n::IntegerUnion) = quaternion_group(PcGroup, n)
 
-function quaternion_group(::Type{T}, n::IntegerUnion) where T <: GAPGroup
-  # FIXME: resolve naming: dicyclic vs (generalized) quaternion: only the
-  # former should be for any n divisible by 4; the latter only for powers of 2.
-  # see also debate on the GAP side (https://github.com/gap-system/gap/issues/2725)
+quaternion_group(::Type{T}, n::IntegerUnion) where T <: GAPGroup = dicyclic_group(T, n)
+quaternion_group(::Type{T}, n::IntegerUnion) where T <: Union{PcGroup, SubPcGroup} = dicyclic_group(T, n)
```

Comment:
```suggestion
quaternion_group(::Type{T}, n::IntegerUnion) where T <: GAPGroup = dicyclic_group(T, n)
```
Is there a reason that these two are identical? If not, I would suggest to unify them